build.rs

```diff
@@ -1,21 +0,0 @@
-#[cfg(all(
-    any(
-        target_os = "linux",
-        target_os = "dragonfly",
-        target_os = "freebsd",
-        target_os = "netbsd",
-        target_os = "openbsd"
-    ),
-    not(feature = "x11"),
-    not(feature = "wayland")
-))]
-compile_error!("at least one of the \"x11\"/\"wayland\" features must be enabled");
-
-#[cfg(all(
-    target_arch = "wasm32",
-    not(feature = "web-sys"),
-    not(feature = "stdweb")
-))]
-compile_error!("at least one of the \"web-sys\"/\"stdweb\" features must be enabled");
-
-fn main() {}
```

Comment:
Is it correct  that it fails in particular in cross build from macOS/Windows to Linux/BSD? Everything winit is using coming from `dlopen`, so there shouldn't be any issues wrt those checks if you build from any os presented in those cfgs to other os presented in those cfgs.

Reply:
I believe the problem is compiling for WASM with  default-features=false.
The build script will be built with the linux target, but none of the x11 or wayland feature will be selected.

In fact, the build script should not fail at compile time, but it should fail at run time, and use the std::env::vars to get the current target.

Reply:
Yeah, I agree. builds.rs should check a target we're compiling for and then throw an error. The main point is that I don't want to put those checks in lib.rs, since I'd like all such stuff be in a one place.

Reply:
Oh, yeah, slight correction: it's impossible to build for WASM from Linux *iff* using `default-features = false` and not enabling x11 or wayland. So, it's possible to work around, but still needs fixed because this is both confusing to users and wastes compile time when compiling to WASM.

I'm noticing several places that `compile_error!()` is still used, and redundant `#cfg[]` checks in several places. Would you rather them all be removed in the same commit if I make sure `build.rs` provides the correct checks, or just fix build.rs first and worry about cleanup later?